src/persist/src/metrics.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Implementation-specific metrics for persist blobs and consensus
+
+use deadpool_postgres::Pool;
+use mz_ore::cast::CastFrom;
+use mz_ore::metric;
+use mz_ore::metrics::{ComputedUIntGauge, Counter, IntCounter, MetricsRegistry};
+use std::sync::Weak;
+
+#[derive(Debug, Clone)]
+pub(crate) struct PostgresConsensusMetrics {
+    pub(crate) _connpool_size: ComputedUIntGauge,
+    pub(crate) connpool_acquires: IntCounter,
+    pub(crate) connpool_acquire_time_seconds: Counter,
+}
+
+impl PostgresConsensusMetrics {
+    pub(crate) fn new(registry: &MetricsRegistry, pool: Weak<Pool>) -> Self {
+        Self {
+            _connpool_size: registry.register_computed_gauge(
```

Comment:
instead of a computed gauge, is it possible to have a normal gauge and update it (via the conn pool hooks) only when a connection is created or destroyed?

Reply:
for posterity: there is no connection destroyed hook, unfortunately